classes/subscription/InstitutionalSubscriptionDAO.inc.php

```diff
@@ -549,23 +549,35 @@ function isValidInstitutionalSubscription($domain, $IP, $journalId, $check = SUB
 
 		// Check if domain match
 		if (!empty($domain)) {
+			$isSqlServer = Config::getVar('database', 'ms_sql');
 			$result = $this->retrieve('
 				SELECT	iss.subscription_id 
 				FROM	institutional_subscriptions iss
 					JOIN subscriptions s ON (iss.subscription_id = s.subscription_id)
 					JOIN subscription_types st ON (s.type_id = st.type_id)
-				WHERE	POSITION(UPPER(LPAD(iss.domain, LENGTH(iss.domain)+1, \'.\')) IN UPPER(LPAD(?, LENGTH(?)+1, \'.\'))) != 0
+				WHERE ' . ($isSqlServer ?
+				           'CHARINDEX(UPPER(RIGHT(REPLICATE(\'.\', LEN(iss.domain) + 1) + LEFT(iss.domain, LEN(iss.domain) + 1), LEN(iss.domain) + 1)), RIGHT(REPLICATE(\'.\', LEN(?) + 1) + LEFT(?, LEN(?) + 1), LEN(?) + 1)) != 0' :
```

Comment:
I think this will break PostgreSQL support -- `charindex` doesn't appear to be a valid function in PostgreSQL. Maybe ADODB has an abstraction that returns the appropriate function name for the DBMS in question?

Reply:
This SQL section is pretty complicated for all databases. This is why I had to keep the SQL Server switch over there. As you say, maybe we can look for something in ADODB, but I'm not confident about finding something that would replace this whole portion of code. It can maybe be reworked entirely to avoid a call at those functions.